core/arch/arm/sm/std_smc.c

```diff
@@ -41,6 +41,67 @@ static const TEE_UUID uuid = {
 	{0x98, 0xd2, 0x74, 0xf4, 0x38, 0x27, 0x98, 0xbb},
 };
 
+#include <trace.h>
+
+static unsigned long arm_arch_version(void)
+{
+	return SMCCC_V_1_1;
+}
+
+static unsigned long arm_arch_feature(unsigned long a1)
+{
+	unsigned long func_id = a1;
+
+	switch (func_id) {
+	case ARM_SMCCC_VERSION:
+		return ARM_SMCCC_RET_SUCCESS;
+	default:
+		return ARM_SMCCC_RET_NOT_SUPPORTED;
+	}
+}
+
+static unsigned long __weak arm_arch_soc_id(void)
```

Comment:
I don't think `static` and `__weak` is a useful combination here.

Reply:
Sorry, initially exported and weak, but I lately changed to local functions.
I will clean up  this mess.